src/amber/router/request.cr

```diff
@@ -0,0 +1,20 @@
+require "./**"
+
+class HTTP::Request
+  # This enables optional cookies store processing
+  @cookies_store : Amber::Router::Cookies::Store?
+
+  # Only enable advance cookies store if the Amber::Server.cookie_enabled?
```

Comment:
Why patch HTTP::Request. Couldn't we just do the same thing on context? Instead of setting it here and copying to context and controller?